nms/app/packages/magmalte/app/state/feg/SubscriberState.js

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @flow strict-local
+ * @format
+ */
+
+import type {EnqueueSnackbarOptions} from 'notistack';
+import type {network_id, subscriber_state} from '@fbcnms/magma-api';
+
+import {FetchSubscriberState} from '../lte/SubscriberState';
+import {getServicedAccessNetworks} from '../../components/FEGServicingAccessGatewayKPIs';
+/**
+ * Props passed when initializing Subscriber State.
+ *
+ * @property {network_id} networkId Id of the federation network.
+ * @property {({[string]: federation_gateway}) => void} setSessionState Sets the subscriber session state.
+ * @property {(msg, cfg,) => ?(string | number),} enqueueSnackbar Snackbar to display error.
+ */
+type InitSubscriberStateProps = {
+  networkId: network_id,
+  setSessionState: ({[network_id]: {[string]: subscriber_state}}) => void,
+  enqueueSnackbar?: (
+    msg: string,
+    cfg: EnqueueSnackbarOptions,
+  ) => ?(string | number),
+};
+
+/**
+ * Initializes the Subscriber State with all the sessions that
+ * this federation network services.
+ *
+ * @param {InitSubscriberStateProps} props an object containing the necessary values to initialize the subscriber state
+ */
+export async function InitSubscriberState(props: InitSubscriberStateProps) {
+  const {networkId, setSessionState, enqueueSnackbar} = props;
+  const sessionState = await FetchFegSubscriberState({
+    networkId,
+    enqueueSnackbar,
+  });
+  setSessionState(sessionState);
+}
```

Comment:
Maybe you're planning on using this function more, but I don't see it being very useful here. Just inline the function in `FEGContext.js`, the function call takes 5 lines, and the body of this function is 6 lines, so this isn't really making anything easier to read.

Reply:
Yeah, this function has a potential to include many other things with fetching subscriber info being one of them or the subscriber metrics and the like. So, I created a function so as to help make it easier for anyone who adds those informations have an easy time(I will not use it for the rest of my project). However, I will just put it in `FEGContext.js` file itself, just as you suggested.